scripts/bootstrap_flux.sh

```diff
@@ -0,0 +1,216 @@
+#!/usr/bin/env bash
+# Copyright 2021-2023 Lawrence Livermore National Security, LLC and other
+# AMSLib Project Developers
+#
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+usage="Usage: $(basename "$0") [#NODES] [JSON file] -- Script that bootstrap Flux on NNODES and writes Flux URIs to the JSON file."
+function version { echo "$@" | awk -F. '{ printf("%d%03d%03d%03d\n", $1,$2,$3,$4); }'; }
+
+# the script needs the number of nodes for flux
+FLUX_NODES="$1"
+# JSON configuration for AMS that will get updated by this script
+AMS_JSON="$2"
+FLUX_SERVER="ams-uri.log"
+FLUX_LOG="ams-flux.log"
+# Flux-core Minimum version required by AMS
+MIN_VAR_FLUX="0.45.0"
+
+re='^[0-9]+$'
+if ! [[ $FLUX_NODES =~ $re ]] ; then
+  echo "[$(date +'%m%d%Y-%T')@$(hostname)] ERROR: '$FLUX_NODES' is not a number."
+  echo $usage  
+  exit 1
+fi
+
+if ! [[ -f "$AMS_JSON" ]]; then
+  echo "[$(date +'%m%d%Y-%T')@$(hostname)] Error: $AMS_JSON does not exists."
+  exit 1
+fi
+
+echo "[$(date +'%m%d%Y-%T')@$(hostname)] Launching Flux with $FLUX_NODES nodes"
+echo "[$(date +'%m%d%Y-%T')@$(hostname)] Writing Flux configuration/URIs into $AMS_JSON"
+
+unset FLUX_URI
+export LC_ALL="C"
+export FLUX_F58_FORCE_ASCII=1
+export FLUX_SSH="ssh"
+# Cleanup from previous runs
+rm -f $FLUX_SERVER $FLUX_LOG
+
+if ! [[ -x "$(command -v flux)" ]]; then
+  echo "[$(date +'%m%d%Y-%T')@$(hostname)] Error: flux is not installed."
+  exit 1
+fi
+echo "[$(date +'%m%d%Y-%T')@$(hostname)] flux = $(which flux)"
+
+flux_version=$(version $(flux version | awk '/^commands/ {print $2}'))
+MIN_VAR_FLUX_LONG=$(version ${MIN_VAR_FLUX})
+# We need to remove leading 0 because they are interpreted as octal numbers in bash
+if [[ "${flux_version#00}" -lt "${MIN_VAR_FLUX_LONG#00}" ]]; then
+  echo "[$(date +'%m%d%Y-%T')@$(hostname)] Error Flux $(flux version | awk '/^commands/ {print $2}') is not supported.\
+  AMS requires flux>=${MIN_VAR_FLUX}"
+  exit 1
+fi
+
+echo "[$(date +'%m%d%Y-%T')@$(hostname)] flux version"
+flux version
+
+# We create a Flux wrapper around sleep on the fly to get the main Flux URI
+FLUX_SLEEP_WRAPPER="./$(mktemp flux-wrapper.XXXXXX).sh"
+cat << 'EOF' > $FLUX_SLEEP_WRAPPER
+#!/usr/bin/env bash
+echo "ssh://$(hostname)$(flux getattr local-uri | sed -e 's!local://!!')" > "$1"
+sleep inf
+EOF
+chmod u+x $FLUX_SLEEP_WRAPPER
+
+MACHINE=$(echo $HOSTNAME | sed -e 's/[0-9]*$//')
+if [[ "$MACHINE" == "lassen" ]] ; then
+  # To use module command we must source this file
+  # Those options are needed on IBM machines (CORAL)
+  # Documented: https://flux-framework.readthedocs.io/en/latest/tutorials/lab/coral.html
+  source /etc/profile.d/z00_lmod.sh
+  module use /usr/tce/modulefiles/Core
+  module use /usr/global/tools/flux/blueos_3_ppc64le_ib/modulefiles
+  module load pmi-shim
+
+  PMIX_MCA_gds="^ds12,ds21" \
+    jsrun -a 1 -c ALL_CPUS -g ALL_GPUS -n ${FLUX_NODES} \
+      --bind=none --smpiargs="-disable_gpu_hooks" \
```

Comment:
-disable_gpu_hooks can disable gpudirect rdma. Leave it as is. I am just bringing it up, cause if we enable load balancing this will break the code.